helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl

```diff
@@ -98,6 +98,27 @@ processors:
   {{- end }}
 
   {{- if and $clusterReceiver.eventsEnabled (eq (include "splunk-otel-collector.logsEnabled" .) "true") }}
+
+  # Add k8s event attributes - k8s.<kind>.name and k8s.<kind>.uid
+  transform/k8sevents:
+    error_mode: ignore
+    log_statements:
+      - conditions:
+          - resource.attributes["k8s.object.kind"] == "HorizontalPodAutoscaler"
+        statements:
+          - set(resource.attributes["k8s.hpa.name"], resource.attributes["k8s.object.name"])
+          - set(resource.attributes["k8s.hpa.uid"], resource.attributes["k8s.object.uid"])
+      - conditions:
+          - resource.attributes["k8s.object.kind"] != "HorizontalPodAutoscaler"
+        statements:
+          - set(resource.attributes[Concat(["k8s", ConvertCase(resource.attributes["k8s.object.kind"], "lower"), "name"], ".")], resource.attributes["k8s.object.name"])
+          - set(resource.attributes[Concat(["k8s", ConvertCase(resource.attributes["k8s.object.kind"], "lower"), "uid"], ".")], resource.attributes["k8s.object.uid"])
+      - conditions:
+          - resource.attributes["k8s.object.kind"] == "Pod" and IsMatch(resource.attributes["k8s.object.fieldpath"], "spec\\.containers.*")
+        statements:
+          - merge_maps(resource.cache, ExtractPatterns(resource.attributes["k8s.object.fieldpath"], "spec.containers\\{(?P<k8s_container_name>[^\\}]+)\\}"), "insert")
```

Comment:
named capture group errors when `.` are used, so I am writing to cache and then setting the attribute in next step